ui/tests/__tests__/App.test.js

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import App from "../../src/App";
+import { mockFacetsGet } from "../../src/api/src/api/FacetsApi";
+import { mockDatasetGet } from "../../src/api/src/api/DatasetApi";
+
+jest.mock("../../src/api/src/api/DatasetApi");
+jest.mock("../../src/api/src/api/FacetsApi");
+("use strict");
+
+beforeEach(() => {
+  mockFacetsGet.mockClear();
+  mockDatasetGet.mockClear();
+});
+
+test("Renders the app empty before receiving all data", () => {
+  mockDatasetGet.mockImplementationOnce(() => {});
+  const tree = shallow(<App />);
+
+  expect(tree).toMatchSnapshot();
+  expect(mockFacetsGet).toHaveBeenCalledTimes(1);
+  expect(mockDatasetGet).toHaveBeenCalledTimes(1);
+});
+
+test("Renders the app after receiving data", () => {
```

Comment:
Can this be the same string for all components (including ExportFab)? Something like "Basic rendering".